GraphAlgorithms/Graph.cs

```diff
@@ -2,38 +2,71 @@
 
 public class Graph
 {
-    public int[,] AdjacencyMatrix { get; set; }
+    public int[][] AdjacencyMatrix { get; set; }
 
     public int Size { get; }
 
     public Graph(int size)
     {
         Size = size;
-        AdjacencyMatrix = new int[size, size];
+        AdjacencyMatrix = new int[size][];
+        for(int i = 0; i < Size; ++i)
+            AdjacencyMatrix[i] = new int[size];
     }
     
-    public void AddEdge(int from, int to) => AdjacencyMatrix[from, to] += 1;
+    public void AddEdge(int from, int to) => AdjacencyMatrix[from][to] += 1;
+
+    public void AddBothSidesEdge(int u, int v)
+    {
+        AdjacencyMatrix[u][v] += 1;
+        AdjacencyMatrix[v][u] += 1;
+    }
     
-    public void SetEdge(int from, int to, int value) => AdjacencyMatrix[from, to] = value;
+    public void SetEdge(int from, int to, int value) => AdjacencyMatrix[from][to] = value;
     
-    public int GetEdge(int from, int to) => AdjacencyMatrix[from, to];
+    public int GetEdge(int from, int to) => AdjacencyMatrix[from][to];
+
+    public int GetNumberOfEdges()
+    {
+        int numOfEdges = 0;
+        foreach(var tab in AdjacencyMatrix)
+        {
+            foreach (var val in tab)
+                numOfEdges += val;
+        }
+        return numOfEdges;
+    }
     
     public int GetDegree(int vertex)
```

Comment:
I have changed the AreNeighbors method to AreNeighborsInBothDirection, because it better fits what method does. I think that GetDegree should return what it returns now. If necessary, GetNumberOfEdgesInBothDirections(int vertex) method or something like this can be added.